docs/root/api-docs/xds_protocol.rst

```diff
@@ -434,10 +440,11 @@ ACK and NACK semantics summary
   received from the management server. The :ref:`response_nonce
   <envoy_v3_api_field_service.discovery.v3.DiscoveryRequest.response_nonce>` field tells the server which of its responses
   the ``ACK`` or ``NACK`` is associated with.
-- ``ACK`` signifies successful configuration update and contains the
+- ``ACK`` signifies that resources were valid and accepted. It contains the
   :ref:`version_info <envoy_v3_api_field_service.discovery.v3.DiscoveryResponse.version_info>` from the
-  :ref:`DiscoveryResponse <envoy_v3_api_msg_service.discovery.v3.DiscoveryResponse>`.
-- ``NACK`` signifies unsuccessful configuration and is indicated by the presence of the
+  :ref:`DiscoveryResponse <envoy_v3_api_msg_service.discovery.v3.DiscoveryResponse>`. This does not
+  mean that the configuration was applied.
+- ``NACK`` signifies that the resources are rejected and is indicated by the presence of the
```

Comment:
I think this is a little misleading.  A NACK means that at least one of the resources in the response was considered invalid, but that does not necessarily mean that none of the resources in the response were accepted -- if a response contains some valid and some invalid resources, the client will NACK, but it may still accept the valid resources from the response.